color_helper.py

```diff
@@ -1001,15 +1001,30 @@ def is_enabled(self, mode, palette_name=None, color=None, auto=False):
         )
 
 
-class ColorHelperFileIndexCommand(sublime_plugin.TextCommand):
-    """Color Helper file index command."""
+class ColorHelperFileIndexCommand(sublime_plugin.WindowCommand):
+    """Forces Color Helper to index the colors in the current file.
 
-    def run(self, edit):
+    If `scan_scopes` argument is provided, instead of using the normal
+    configuration it will look for colors inside the given scopes.
+    """
+
+    def run(self, scan_scopes=None, allowed_colors=["all"], use_hex_argb=False):
         """Run the command."""
-        rules = util.get_rules(self.view)
-        if rules and util.get_scope(self.view, rules, skip_sel_check=True):
+        view = self.window.active_view()
+        if scan_scopes:
+            if "all" in allowed_colors:
+                allowed_colors = util.ALL
+            rules = view.settings().get('color_helper.scan')
```

Comment:
Will this setting exist always? No idea about the life cycle of the plugin, but it seems reasonable to return a `{}` default here.

Reply:
Agreed, there should be a default.

Reply:
this is normally always set because `set_file_scan_rules` set it every time a view is selected.
I added a default, but note that if the "color_helper.scan" wasn't set before then `set_file_scan_rules` will erase the value I set the next time it runs.